#!/usr/bin/env python
PACKAGE = "ouster_flir_calibration"
from dynamic_reconfigure.parameter_generator_catkin import *
gen = ParameterGenerator()

# extract_lidar_points.launch
gen.add("min_x", double_t , 0, "min_x", +2.80, -20.0, 20.0)
gen.add("max_x", double_t , 0, "max_x", +4.00, -20.0, 20.0)
gen.add("min_y", double_t , 0, "min_y", -0.70, -20.0, 20.0)
gen.add("max_y", double_t , 0, "max_y", +0.80, -20.0, 20.0)
gen.add("min_z", double_t , 0, "min_z", -1.00, -20.0, 20.0)
gen.add("max_z", double_t , 0, "max_z", +0.35, -20.0, 20.0)

# result.launch
gen.add("xlim_low",  double_t , 0, "xlim_low",    +0.00, -200.0, 200.0)
gen.add("xlim_high", double_t , 0, "xlim_high", +100.00, -200.0, 200.0)
gen.add("ylim_low",  double_t , 0, "ylim_low",  -200.00, -200.0, 200.0)
gen.add("ylim_high", double_t , 0, "ylim_high", +200.00, -200.0, 200.0)
gen.add("zlim_low",  double_t , 0, "zlim_low",   -50.00, -200.0, 200.0)
gen.add("zlim_high", double_t , 0, "zlim_high",  +50.00, -200.0, 200.0)

exit(gen.generate(PACKAGE, "dynamic_reconfigure", "dyn_reconfig"))
